src/libraries/illink-sharedframework.targets

```diff
@@ -28,6 +28,7 @@
       <LinkerNoWarn Condition="'$(TargetOS)' != 'windows'">$(LinkerNoWarn);IL2008</LinkerNoWarn>
       <LinkerNoWarn Condition="'$(Platform)' != 'x64' AND '$(Platform)' != 'arm64'">$(LinkerNoWarn);IL2012</LinkerNoWarn>
       <ILLinkArgs>$(ILLinkArgs) --nowarn $(LinkerNoWarn)</ILLinkArgs>
+      <ILLinkArgs>$(ILLinkArgs) --generate-warning-suppressions xml</ILLinkArgs>
```

Comment:
Should we always do this? Or should it be behind some flag?

Reply:
It makes sense for this to be opt-in only. I'll push another commit.